src/03-ticket-calculator.js

```diff
@@ -54,7 +55,134 @@ const exampleTicketData = require("../data/tickets");
+
+//       return `Ticket type '${ticketInfo[i].ticketType}' cannot be found.`
+
+//     } else if (!ticketData[ticketInfo[i].ticketType].priceInCents.hasOwnProperty(ticketInfo[i].entrantType )){
+
+//       return `Entrant type '${ticketInfo[i].entrantType}' cannot be found.`
+
+//     }
+
+//     result += totalPrice
+
+//   }
+
+//   return result
+
+// }
+  if (!Object.keys(ticketData).includes(ticketInfo.ticketType)){
+    return `Ticket type '${ticketInfo.ticketType}' cannot be found.`
+  }
+  if (!Object.keys(ticketData.general.priceInCents).includes(ticketInfo.entrantType)){
+    return `Entrant type '${ticketInfo.entrantType}' cannot be found.`
+  }
+  let ticketPrice = 0;
+switch (ticketInfo.ticketType) {
+  case 'general':
+    switch (ticketInfo.entrantType) {
+      case 'child':
+        ticketPrice += ticketData.general.priceInCents.child;
+        break;
+      case 'adult':
+        ticketPrice += ticketData.general.priceInCents.adult;
+        break;
+      case 'senior':
+        ticketPrice += ticketData.general.priceInCents.senior;
+        break;
+    }
+    break;
+  case 'membership':
+    switch (ticketInfo.entrantType) {
+      case 'child':
+        ticketPrice += ticketData.membership.priceInCents.child;
+        break;
+      case 'adult':
+        ticketPrice += ticketData.membership.priceInCents.adult;
+        break;
+      case 'senior':
+        ticketPrice += ticketData.membership.priceInCents.senior;
+        break;
+    }
+    break;
+    }
+for (let i = 0; i < ticketInfo.extras.length; i++){
+  switch (ticketInfo.extras[i]){
+    case 'movie':
+      switch (ticketInfo.entrantType) {
+        case 'child':
+          ticketPrice += ticketData.extras.movie.priceInCents.child;
+          break;
+        case 'adult':
+          ticketPrice += ticketData.extras.movie.priceInCents.adult;
+          break;
+        case 'senior':
+          ticketPrice += ticketData.extras.movie.priceInCents.senior;
+          break;
+      }
+    break;
+    case 'education':
+      switch (ticketInfo.entrantType) {
+        case 'child':
+          ticketPrice += ticketData.extras.education.priceInCents.child;
+          break;
+        case 'adult':
+          ticketPrice += ticketData.extras.education.priceInCents.adult;
+          break;
+        case 'senior':
+          ticketPrice += ticketData.extras.education.priceInCents.senior;
+          break;
+      }
+    break;
+    case 'terrace':
+      switch (ticketInfo.entrantType) {
+        case 'child':
+          ticketPrice += ticketData.extras.terrace.priceInCents.child;
+          break;
+        case 'adult':
+          ticketPrice += ticketData.extras.terrace.priceInCents.adult;
+          break;
+        case 'senior':
+          ticketPrice += ticketData.extras.terrace.priceInCents.senior;
+          break;
+      }
+    break;
+    default:
+      return `Extra type '${ticketInfo.extras[i]}' cannot be found.`
```

Comment:
This code works but it can definitely be optimized and made more efficient. In instances like this you want to leverage your variables to check if entrantType, ticketType, or extras exist within the the `ticketData` object. For instance if you are checking the `ticketType ` that might look like 
```
if( !ticketData[ticketInfo.ticketType] ){ // if it is falsey then it is safe to say the ticketType provided is incorrect 
   return `Ticket type '${ticketInfo.ticketType}' cannot be found.`;
}
```
And getting the price for the `entrantType` might look like `const price = ticketData[ticketInfo.ticketType].priceInCents[ticketInfo.entrantType];`